aten/src/ATen/native/cuda/MemoryAccess.cuh

```diff
@@ -350,11 +350,22 @@ inline C10_HOST_DEVICE int can_vectorize_up_to(const char *pointer) {
   uint64_t address = reinterpret_cast<uint64_t>(pointer);
   constexpr int vec2_alignment = std::alignment_of<aligned_vector<scalar_t, 2>>::value;
   constexpr int vec4_alignment = std::alignment_of<aligned_vector<scalar_t, 4>>::value;
-  if (address % vec4_alignment == 0) {
-    return 4;
-  } else if (address % vec2_alignment == 0) {
-    return 2;
-  }
+  #if defined(USE_ROCM)
+    constexpr int vec8_alignment = std::alignment_of<aligned_vector<scalar_t, 8>>::value;
```

Comment:
Nit:

Might be better readability and code segment.

```
  constexpr int vec2_alignment = std::alignment_of<aligned_vector<scalar_t, 2>>::value;
  constexpr int vec4_alignment = std::alignment_of<aligned_vector<scalar_t, 4>>::value;
#ifdef ROCM
  constexpr int vec8_alignment = std::alignment_of<aligned_vector<scalar_t, 8>>::value;
  if (address % vec8_alignment == 0) {
    return 8;
  }
#endif
  if (address % vec4_alignment == 0) {
    return 4;
  } else if (address % vec2_alignment == 0) {
    return 2;
  }
```

Here, `if` vs `else if`, it's fine because the only statement in the `if` block is a `return`.